From 5ede3f8aed9a1a579bf7304142600d1f3500add9 Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Mon, 12 Jun 2023 03:58:42 +0100
Subject: [PATCH 2/2] init: bypass device lookup for /dev/fit* rootfs

Allow 'rootwait' as /dev/fit* can show up late if the underlaying
device is probed late.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 init/do_mounts.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/init/do_mounts.c
+++ b/init/do_mounts.c
@@ -645,7 +645,8 @@ void __init prepare_namespace(void)
 
 	if (saved_root_name[0]) {
 		root_device_name = saved_root_name;
-		if (!strncmp(root_device_name, "mtd", 3) ||
+		if (!strncmp(root_device_name, "fit", 3) ||
+		    !strncmp(root_device_name, "mtd", 3) ||
 		    !strncmp(root_device_name, "ubi", 3)) {
 			mount_block_root(root_device_name, root_mountflags);
 			goto out;
